tests/Credentials/AssumeRoleWithWebIdentityCredentialProviderTest.php

```diff
@@ -396,4 +397,85 @@ public function testCanDisableInvalidIdentityTokenRetries()
             unlink($tokenPath);
         }
     }
+
+    /**
+     * @dataProvider endpointConfigurationProvider
+     */
+    public function testEndpointConfigurationBasedOnRegion(
+        string $region,
```

Comment:
We need to make this parameter nullable to fix the test failure, but overall looks good.

Reply:
ah, went back and added types and just pushed without checking. fixed now